scripts/apps/authoring/authoring/services/AuthoringService.ts

```diff
@@ -368,7 +368,14 @@ export function AuthoringService($q, $location, api, lock, autosave, confirm, pr
         helpers.filterDefaultValues(extDiff, orig);
         var endpoint = 'archive_' + action;
 
-        return api.update(endpoint, orig, extDiff, {publishing_warnings_confirmed: publishingWarningsConfirmed})
+        var params = {
+            publishing_warnings_confirmed: publishingWarningsConfirmed,
+            desk_id: appConfig.features.publishFromPersonal && !orig?.task?.desk && !extDiff?.task?.desk
```

Comment:
If `desk_id` is only needed when publishing from personal, can you only add it to params only in that case?

```
if (publishFromPersonal) {
    params.desk_id = ...;
}
```